src/dmd/cli.d

```diff
@@ -582,6 +582,23 @@ dmd -cov -unittest myprog.d
             If $(I libname) is empty, no C runtime library is automatically linked in.",
             TargetOS.Windows,
         ),
+        Option("mtriple=<triple>",
+               "use <triple> as <arch>-[<vendor>-]<os>[-<cenv>[-<cppenv]]",
+               "$(I arch) is the architecture: either `x86`, `x64`, `x86_64` or `x32`,
+               $(I vendor) is optional and always ignored,
+               $(I os) is the operating system, this may have a trailing version number (required for freebsd):
+               `freestanding` for no operating system,
+               `darwin` or `osx` for MacOS, `dragonfly` or `dragonflybsd` for DragonflyBSD,
+               `freebsd`, `openbsd`, `linux`, or `windows` for their respective operating systems.
+               $(I cenv) is the C runtime environment and is optional: `musl` for musl-libc,
+               `msvc` or `microsoft` for the MSVC runtime (the default for windows with this option),
+               `bionic` for the andriod libc, `dm` or `digital_mars` for the digital mars runtime for Windows
+               `gnu` or `glibc` for the GCC C runtime, `newlib`, `uclibc` or `wasi` for their respective C runtimes.
+               ($ I cppenv) is the C++ runtime environment: `clang` for the LLVM C++ runtime
+               `gcc` for GCC's C++ runtime, `msvc` or `microsoft` for microsoft's MSVC C++ runtime (the default for windows with this switch),
+               `sun` for Sun's C++ runtime and `dm` or `digital_mars` for the Digital Mars C++ runtime for windows.
```

Comment:
Need an update to conditional compilation, ideally by extending [this](https://dlang.org/spec/version.html#predefined-versions) to show which triple matches which version (and platform's defaults).

Reply:
Ehh, I'd rather not. `-target` is very compiler specific and dmd doesn't support the vast majority of things that could be set by it, also LDC uses a different switch name (`-mtriple`)